cip-0016.md

```diff
@@ -0,0 +1,127 @@
+
+When validating assets, in addition to existing checks, the given asset should be checked to ensure:
+- The asset has been locked.
+
+### Source Address
+
+Normal dividends exclude the source address, as it is unncessary to debit their balance just to immediately credit it again. Their share of a dividend is implicit and is just the amount left in their account after paying other holders. Scheduled dividends will not exclude the source address, as it is necessary, in this instance.
+
+### Activation Index
+
+New consensus variables known as ```MIN_ACTIVATION``` and ```MAX_ACTIVATION``` will be used to instate an upper and lower limit on dividend lockup times. The corresponding value for order expirations is ```MAX_EXPIRATION = 4 * 2016   # Two months```.
+
+This CIP defines ```MIN_ACTIVATION = 144 # One day``` and ```MAX_ACTIVATION = 210000 # Four years```.
+
+Imposing a one day minimum helps avoid the edge case where a transaction confirms after the scheduled activation index, which would act no differently than a normal dividend, for potentially a higher fee gas fee. And having a reasonably distant upper limit keeps scheduled dividends from imposing too much cost on the network.
+
+When validating activation indexes, the given value should be checked to ensure:
+- The activation index is an integer.
+- The activation index is unsigned.
+- The activation index is a future block height.
+- The difference between the current block height and the activation index is not less than ```MIN_ACTIVATION```.
+- The difference between the current block height and the activation index is not greater than ```MAX_ACTIVATION```.
+
+To store the activation index, tables need to be altered and created as follows (closely follows order expirations)...
+
+```
+---Pseudocode---
+ALTER TABLE dividends
+ADD COLUMN activate_index INTEGER
+CREATE INDEX IF NOT EXISTS
+activate_idx ON dividends (activate_index, status)
+
+CREATE TABLE IF NOT EXISTS dividend_activations
+dividend_index INTEGER PRIMARY KEY,
+dividend_hash TEXT UNIQUE,
+source TEXT,
+block_index INTEGER,
+FOREIGN KEY (block_index) REFERENCES blocks(block_index),
+CREATE INDEX IF NOT EXISTS
+block_index_idx ON dividend_activations (block_index)
+CREATE INDEX IF NOT EXISTS
+source_idx ON dividend_activations (source)
+
+```
+
+After this change, dividends can be differentiated from scheduled dividends by using the dividend_activations table, looking for the 'pending' status, or looking for the presence of an activation index.
+
+### Message Encoding
+
+At present, there are two encoding formats defined...
+```
+FORMAT_1 = '>QQ'
+LENGTH_1 = 8 + 8
+FORMAT_2 = '>QQQ'
+LENGTH_2 = 8 + 8 + 8
+```
+
+This CIP would created a third ```FORMAT_3``` such that...
+```
+FORMAT_3 = '>QQQH'
```

Comment:
Why an H for an index? is index just a block delta? If so, that would allow only 1.24 years time difference.... depending on the MAX_EXPIRATION value this should be adjusted